### ISSUE TITLE
[iOS] AdHoc IPA app builds cannot be installed with AIR 33.1.1.406

### ISSUE THREAD

**Issue by itlancer** [Gamua/Adobe-Runtime-Support#708]
## Problem Description
AdHoc IPA application builds compiled with AIR 33.1.1.406 cannot be installed to iOS devices. So you cannot debug or distribute application for iOS.

It has been tested with latest AIR 33.1.1.406 with many different iOS devices with different applications (even "empty").
Tried to install it via USB and Diawi.
Same problem in all cases.

It works fine with AIR 33.1.1.345 and below with the same applications.

## Steps to Reproduce
Compile any application (even "empty") with AIR 33.1.1.406 and try to install it on any iOS device.
Command line for app compilation looks like:
`adt.bat -package -target ipa-ad-hoc -storetype pkcs12 -keystore cert.p12 -storepass pass -keypass pass -provisioning-profile profile.mobileprovision build.ipa descriptor.xml -extdir libs App.swf`

**Actual Result:**
Application will not be installed.
You will get error:
```
WARNING: could not locate iTunesMetadata.plist in the archive!
WARNING: could not locate Payload/app.id.app/SC_Info/app.id.sinf in archive!
ERROR: Install failed. Got error "ApplicationVerificationFailed" with code 0xe8008001: Failed to verify code signature of /private/var/installd/Library/Caches/com.apple.mobile.installd.staging/temp.59Zx1g/extracted/Payload/app.id.app : 0xe8008001 (An unknown error has occured)
```

**Expected Result:**
Application will be installed and could work.

## Known Workarounds
Use AIR SDK 33.1.1.345 or patch from here: https://github.com/Gamua/Adobe-Runtime-Support/issues/511#issuecomment-785492035

**Comment by ladigitale** [comment]
Using SDK 33.1.1.345 didn't work form me. 
As suggested in the workarounds the patch purposed in #511 worked.

**Comment by kasparv** [comment]
For me, after-compile signing workaround described in #676 works for AppStore.

**Comment by YasukazuU** [comment]
I received the same error.
In my case, it was because I didn't have itunes on my mac.
The reason was that I was using a mac os after Catalina.

**Comment by megharajdeepak** [comment]
Hi,
I wanted to use latest air version ie.., 33.1.1.406 to fix the signing issue.
However, after compiling the application and on click of Finish, I get a message saying 'Releasing export build..'. But after few seconds, the same screen shows up again without any progress and ipa is not getting generated even after 4 hours:
![image](https://user-images.githubusercontent.com/5477084/110765899-a4f3e380-827a-11eb-9a98-45f77c2d5581.png)

**Comment by ajwfrost** [comment]
Hi

We have heard of an issue in Flash Builder that's causing this problem - it happened too with our very first 31.0 build.. Assuming this only happens the second/subsequent times you run this after you start up Flash Builder, then it's probably this same issue, an internal error in the way Flash Builder orders the libraries before passing them to the compiler.

It seems a little spurious, and it may well be that our next release just works whereas this .406 build seems to have an issue. 
Although Flash Builder isn't actually supported any more, I will see whether we can try to find a resolution here..

thanks

**Comment by MarkBurvs** [comment]
Getting this exact same error while trying to package for iOS using IntelliJ IDEA. 
AIR version 33.1.1.406.

**Comment by megharajdeepak** [comment]
@ajwfrost Ok, kindly let me know if there is any other way to generate IPA using latest air SDK. May be through Command line? I have p12 certificate and it's password.

**Comment by megharajdeepak** [comment]
@ajwfrost  Basically, the users are facing this issue:

![image](https://user-images.githubusercontent.com/5477084/110888486-4d08bb80-8312-11eb-9e84-e09c78e546f0.png)

Let me know which stable version of AIR I use to resolve that.

**Comment by Ekitty** [comment]
Getting this exact same error while trying to package for iOS using IntelliJ IDEA.
AIR version 33.1.1.406.

**Comment by ajwfrost** [comment]
@megharajdeepak  what version of the AIR SDK did you use to create that version of the application?

@Ekitty sorry can you please clarify, which is the error you're seeing? This thread contains a few different things.. and are you using just the 'raw' 33.1.1.406 build, or did you patch it with the above updated ADT.jar file?

thanks

**Comment by megharajdeepak** [comment]
@ajwfrost  Latest version with which the app was compiled and built was by using  Flex 4.14.1 Flash Player 17.0 AIR 18.
And recently users started getting the above message after they updated their iOS version to 14.3 and above. 

And hence, I tried to recompile the same app both with Flex_4.14.1_AIR_33.1.1.406_FP17 as well as Flex_4.14.1_AIR_33.1.1.345_FP17. But same issue in both versions. Build is not created and there are not errors. I am using Adobe Flash Builder 4.6.

**Comment by ajwfrost** [comment]
@megharajdeepak thanks - so the issue the end users are seeing is then most likely just because of the very old AIR runtime in there (32-bit perhaps?) ... once we've resolved these other issues then it hopefully means you can use the next AIR SDK to rebuild/re-package the application and it should start working without warnings..

thanks

**Comment by megharajdeepak** [comment]
@ajwfrost Perhaps. But I am not quite sure what that mean and what needs to be addressed, as that warning message is thrown as soon as we open the installed app on iPad. Just to make it clear, this warning is coming from the iOS and not the flex\AIR app. I am just worried that from one of the up coming iOS versions, we may not get 'OK' button to proceed and apps will not work. 

If that is addressed in next AIR SDK, that will surely help. Thank you so much.

**Comment by marchbold** [comment]
This could be an issue with using an older certificate as well, have you updated your certificates lately?

**Comment by FliplineStudios** [comment]
Just wanted to chime in that the popup @megharajdeepak shared is also appearing in 33.1.1.345 adhoc builds and is more related to the separate issue with the new code signature format seen in Issue #590 .  For an App Store release, Apple will replace the signature anyway once uploaded so we didn't have to worry about our end users seeing that popup, but for adhoc/debug it is indeed an ongoing issue.

**Comment by zishani** [comment]
Hi all,

We can confirm that 13.1.1.406 does **not** resolve the Enterprise In House certificate distribution issues due to the new signing requirements.

But I notice that Harman have acknowledged this in the release notes:

`updating IPA CodeResources signature format (work in progress for Gamua-590)`

I assume the next release will fix this - as currently we have to resign IPAs after building.....

**Comment by ajwfrost** [comment]
Thanks all. So .. the "will not work with future versions of iOS" is an ambiguous message and we perhaps can't be 100% sure why this is appearing, there could be multiple reasons! But the expectation is that if you use our next SDK release, which fixes the certificate and code signing version issues, then these should all go away!

**Comment by itlancer** [comment]
For my cases it fixed with latest AIR SDK 33.1.1.444. Thanks!
If someone still experienced similar issues please create new ticket.
